plugins/woocommerce/src/Blocks/AIContent/UpdateProducts.php

```diff
@@ -321,13 +321,7 @@ private function product_image_upload( $product_id, $image_src, $image_alt ) {
 		set_time_limit( 150 );
 		wp_raise_memory_limit( 'image' );
 
-		$product_image_id = media_sideload_image( $image_src, $product_id, $image_alt, 'id' );
-
-		if ( is_wp_error( $product_image_id ) ) {
-			return $product_image_id->get_error_message();
-		}
```

Comment:
Sure! Before we were checking if the `$product_image_id` was an error and returning just the error message as a string making this error checking useless 👉 https://github.com/woocommerce/woocommerce/blob/49948c8c6f4c84048f0ded3f5afc58643ff1d5cf/plugins/woocommerce/src/Blocks/AIContent/UpdateProducts.php#L196 since the return from `product_image_upload` was never a `WP_Error` but a string .

With this change, we return whatever the `media_sideload_image` returns, so we make possible for whoever is calling `product_image_upload` to handle the error. Does it make sense?

Reply:
That makes sense. Thank you for taking the time to explain 🙏🏻